kornia/augmentation/functional.py

```diff
@@ -0,0 +1,332 @@
+    """
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
+        batch_size = input.shape[0] if len(input.shape) == 4 else 1
+    params = pg._random_prob_gen(batch_size, p=p)
+    return apply_vflip(input, params, return_transform)
+
+
+def color_jitter(input: torch.Tensor, brightness: FloatUnionType = 0.,
+                 contrast: FloatUnionType = 0., saturation: FloatUnionType = 0.,
+                 hue: FloatUnionType = 0., return_transform: bool = False) -> UnionType:
+    r"""Generate params and apply operation on input tensor.
+
+    See :func:`~kornia.augmentation.param_gen._random_color_jitter_gen` for details.
+    See :func:`~kornia.augmentation.functional.apply_color_jitter` for details.
+    """
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
+        batch_size = input.shape[0] if len(input.shape) == 4 else 1
+    params = pg._random_color_jitter_gen(batch_size, brightness, contrast, saturation, hue)
+    return apply_color_jitter(input, params, return_transform)
+
+
+def random_grayscale(input: torch.Tensor, p: float = 0.5, return_transform: bool = False):
+    r"""Generate params and apply operation on input tensor.
+
+    See :func:`~kornia.augmentation.param_gen._random_prob_gen` for details.
+    See :func:`~kornia.augmentation.functional.apply_grayscale` for details.
+    """
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
+        batch_size = input.shape[0] if len(input.shape) == 4 else 1
+    params = pg._random_prob_gen(batch_size, p=p)
+    return apply_grayscale(input, params, return_transform)
+
+
+def apply_hflip(input: torch.Tensor, params: Dict[str, torch.Tensor], return_transform: bool = False) -> UnionType:
+    r"""Apply Horizontally flip on a tensor image or a batch of tensor images with given random parameters.
+    Input should be a tensor of shape (H, W), (C, H, W) or a batch of tensors :math:`(*, C, H, W)`.
+
+    Args:
+        params (dict): A dict that must have {'batch_prob': torch.Tensor}. Can be generated from
+        kornia.augmentation.param_gen._random_prob_gen.
+        return_transform (bool): if ``True`` return the matrix describing the transformation applied to each
+        input tensor.
+
+    Returns:
+        torch.Tensor: The horizontally flipped input
+        torch.Tensor: The applied transformation matrix :math: `(*, 3, 3)` if return_transform flag
+        is set to ``True``
+    """
+
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if len(input.shape) == 2:
+        input = input.unsqueeze(0)
+
+    if len(input.shape) == 3:
+        input = input.unsqueeze(0)
+
+    if len(input.shape) != 4:
+        raise ValueError(f"Input size must have a shape of (*, C, H, W). Got {input.shape}")
```

Comment:
I was struggling with this one previously.

In a way, I think it is better to make it clear that this function accepts only (H, W), (C, H, W) and (B, C, H, W) than any arbitrary shapes have more than 2 dimensions. Let's say that if a user accidentally wanna input a bunch of 3D volumetric images with shape (B, C, H, W, D), the operation will not be processed unless they reshape them to be "processable".

Also, if this function accepts an arbitrary shape of (..., K, B, C, H, W), I think it makes more sense to reshape it back to what it was than returning (B, C, H, W) only.

For now, I prefer sticking to what we have implemented.

Reply:
Well, I guess this will have to be discussed but keep in mind that the docstring specifies input of shape (*, C, H, W) and not (B, C, H, W).  So in the case for volumetric images if the user inputs a tensor of shape  (B, C, H, W, D) he would not be following the docs. 

Nevertheless for now you leave your approach, however if you do in fact decide to so I think the lines:
```
input = input.unsqueeze(0)
input = input.view((-1, (*input.shape[-3:])))
```
should be removed from every apply function and the docstring modified appropriatly. @edgarriba thoughts?

Reply:
I think @dvd42 is right, for len > 4 we can arrange the data in that way, then we have to keep in mind to reshape in the original format before returning the processed data